fvm/context.go

```diff
@@ -393,11 +383,34 @@ func WithEVMTracer(tracer debug.EVMTracer) Option {
 	}
 }
 
-// WithReadVersionFromNodeVersionBeacon sets whether the version from the node version beacon should be read
-// this should only be disabled for testing
-func WithReadVersionFromNodeVersionBeacon(enabled bool) Option {
+// WithEntropyProvider sets the entropy provider of a virtual machine context.
+//
+// The VM uses the input to provide entropy to the Cadence runtime randomness functions.
+func WithEntropyProvider(source environment.EntropyProvider) Option {
+	return func(ctx Context) Context {
+		ctx.EntropyProvider = source
+		return ctx
+	}
+}
+
+// WithExecutionVersionProvider sets the execution version provider of a virtual machine context.
+//
+// this is used to provide the execution version to the Cadence runtime.
+func WithExecutionVersionProvider(provider environment.ExecutionVersionProvider) Option {
+	return func(ctx Context) Context {
+		ctx.ExecutionVersionProvider = provider
+		return ctx
+	}
+}
+
+// WithProtocolStateSnapshot sets all the necessary components from a subset of the protocol state
+// to the virtual machine context.
+func WithProtocolStateSnapshot(snapshot flow.ProtocolSnapshotExecutionSubset) Option {
 	return func(ctx Context) Context {
-		ctx.ReadVersionFromNodeVersionBeacon = enabled
+
+		ctx = WithEntropyProvider(snapshot)(ctx)
```

Comment:
better to add some comments explain how it works for the two lines.

Reply:
The comment is already on the `WithEntropyProvider` and on the `snapshot.RandomSource` I don't think it will help adding it here as well